src/forms/forms.service.ts

```diff
@@ -1,12 +1,36 @@
 import { Injectable } from '@nestjs/common';
 import { TemplateRepository } from 'src/repositories/template.repository';
 import { CreateTemplate } from './dto/create.template';
+import { ListRepository } from 'src/repositories/list.repository';
+
 
 @Injectable()
 export class FormsService {
-  constructor(private readonly templateRepository: TemplateRepository) {}
+  constructor(private readonly templateRepository: TemplateRepository,
+    private readonly listRepository: ListRepository)  {}
 
   async createTemplate(createTemplate: CreateTemplate) {
     return this.templateRepository.createTemplate(createTemplate);
   }
+
+  async getTemplateByName(name: string){
+    const response = await this.templateRepository.getTemplateByName(name)
+    const { question } = response
+    const includesList = question.filter( (question) => question['type'].includes('list') )
+    const lists = [...new Set(includesList.map((question) => question['resource']))]
+
+    const listsArray = await Promise.all( lists.map( (list) => {
+      return this.listRepository.getListByName(list)
+    }))
```

Comment:
You are working in a getTemplateByName scope so, this logic is another responsibility maybe, you should move this logic in an isolated piece of code to improve your readability and the unit test.

Reply:
Moved to manageListsContent function